### DISCUSSION TITLE
EPSON WF-2606 WSD Scan over Device Panel

### DISCUSSION THREAD

**Discussion by baldisos** [alexpevzner/sane-airscan#124] [Q&A]
I'm trying to use my EPSON WF-2660 as a network scanner. It has a Function to scan to WSD-compatible devices directly over it's control panel. It works for WSD-enabled Windows Devices, but i can't seem to get any functionality out of sane-airscan with this. I don't actually want to scan over the PC (or more like a server) itself, i want to scan over the printer controlpanel and have a file saved to disk or a network location in that case. Is there any possibilty to recreate such behavior with this?

**Comment by a2intl** [comment]
I've been looking into this, and it's a rather an involved addition: it requires using the Subscribe WSD action (which includes a list of ScanDestinations) to subscribe to ScanAvailableEvents, and then receiving those ScanAvailableEvents (which requires an HTTP server being started) and firing off the appropriate StartScanJob action when those are received. There is a python program that's been written for this: https://github.com/Tobag/wsd-scan which might meet your needs.
